sdk/tables/azure-data-tables/azure/data/tables/_serialize.py

```diff
@@ -220,33 +206,28 @@ def _add_entity_properties(source):
     # set properties type for types we know if value has no type info.
     # if value has type info, then set the type to value.type
     for name, value in to_send.items():
-        mtype = ""
-
+        if value is None:
+            continue
+        mtype: Optional[EdmType] = None
         if isinstance(value, Enum):
-            try:
-                convert = _PYTHON_TO_ENTITY_CONVERSIONS.get(unicode)  # type: ignore
-            except NameError:
-                convert = _PYTHON_TO_ENTITY_CONVERSIONS.get(str)
+            convert = _PYTHON_TO_ENTITY_CONVERSIONS[str]
             mtype, value = convert(value)
         elif isinstance(value, datetime):
             mtype, value = _to_entity_datetime(value)
-        elif isinstance(value, tuple):
-            convert = _EDM_TO_ENTITY_CONVERSIONS.get(value[1])
+        elif isinstance(value, tuple) and value[0] is not None:
+            convert = _EDM_TO_ENTITY_CONVERSIONS[EdmType(value[1])]
             mtype, value = convert(value[0])
         else:
-            convert = _PYTHON_TO_ENTITY_CONVERSIONS.get(type(value))
-            if convert is None and value is not None:
-                raise TypeError(_ERROR_TYPE_NOT_SUPPORTED.format(type(value)))
-            if value is None:
-                convert = _to_entity_none
-
+            try:
+                convert = _PYTHON_TO_ENTITY_CONVERSIONS[type(value)]
+            except KeyError:
+                raise TypeError(_ERROR_TYPE_NOT_SUPPORTED.format(type(value))) from None
```

Comment:
from KeyError?

Reply:
Nah - the KeyError isn't really relevant to customers here.